python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -688,8 +703,191 @@ def asset_check_spec(self) -> AssetCheckSpec:
+                    f" around I/O managers. If you not using I/O managers we suggest you {alt}. If"
+                    " you are using I/O managers the method still exists at"
+                    f" op_execution_context.{attr}."
+                ),
+                breaking_version="1.7",
+                stacklevel=1,
+            )
+
+        if attr in ALTERNATE_AVAILABLE_METHODS:
+            deprecation_warning(
+                subject=f"AssetExecutionContext.{attr}",
+                additional_warn_text=f"Instead {ALTERNATE_AVAILABLE_METHODS[attr]}.",
+                breaking_version="1.7",
+                stacklevel=1,
+            )
+
+        return getattr(self._op_execution_context, attr)
+
+    # include all supported methods below
+
+    @public
+    @property
+    def op_execution_context(self) -> OpExecutionContext:
+        return self._op_execution_context
+
+    @public
+    @property
+    def run_id(self) -> str:
+        return self._op_execution_context.run_id
+
+    @public
+    @property
+    def dagster_run(self) -> DagsterRun:
+        """PipelineRun: The current pipeline run."""
+        return self._step_execution_context.dagster_run
+
+    @public
+    @property
+    def asset_key(self) -> AssetKey:
+        return self._op_execution_context.asset_key
+
+    @public
+    @property
+    def pdb(self) -> ForkedPdb:
+        return self._op_execution_context.pdb
+
+    @public
+    @property
+    def log(self) -> DagsterLogManager:
+        """DagsterLogManager: The log manager available in the execution context."""
+        return self._op_execution_context.log
+
+    @public
+    # renaming to avoid ambiguity in single run and multi-partition case
+    @property
+    def is_partitioned_execution(self) -> bool:
+        return self._op_execution_context.has_partition_key
+
+    @public
+    def log_event(self, event: UserEvent) -> None:
+        return self._op_execution_context.log_event(event)
+
+    @public
+    @property
+    def assets_def(self) -> AssetsDefinition:
+        return self._op_execution_context.assets_def
+
+    @public
+    # TODO confirm semantics in the presense of asset subsetting
+    # seems like there should be both "asset_keys" and "selected_asset_keys"
+    @property
+    def selected_asset_keys(self) -> AbstractSet[AssetKey]:
+        return self._op_execution_context.selected_asset_keys
+
+    @public
+    @experimental
+    def get_asset_provenance(self, asset_key: AssetKey) -> Optional[DataProvenance]:
+        return self._op_execution_context.get_asset_provenance(asset_key)
+
+    @property
+    def asset_check_spec(self) -> AssetCheckSpec:
+        return self._op_execution_context.asset_check_spec
+
+    @public
+    @property
+    def partition_key_range(self) -> PartitionKeyRange:
+        return self._op_execution_context.asset_partition_key_range
+
+    @public
+    def partition_key_range_for_asset_key(self, asset_key: AssetKey) -> PartitionKeyRange:
```

Comment:
just leaving a note that making the partition methods agnostic to "input" or "output" lead to some pretty complex errors. Some of the methods (like the partition_key_range one) have different code paths for "inputs" and "outputs" because of partition mapping. Not worth trying to do all of that implementation detail in this RFC, but for a final implementation, we should make sure this code path is really thoroughly tested